stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt

```diff
@@ -435,59 +435,38 @@ internal class ChannelControllerImpl(
         val queryOfflineJob = domainImpl.scope.async { runChannelQueryOffline(pagination) }
 
         // start the online query before queryOfflineJob.await
-        val queryOnlineJob = if (domainImpl.isOnline()) {
-            domainImpl.scope.async { runChannelQueryOnline(pagination) }
-        } else {
-            // if we are not offline we mark it as needing recovery
-            recoveryNeeded = true
-            logger.logI("Skipping channel.watch for channel $cid since we are offline. Marking it as needing recovery.")
-            null
-        }
-        val localChannel = queryOfflineJob.await()
-        if (localChannel != null) {
+        val queryOnlineJob = domainImpl.scope.async { runChannelQueryOnline(pagination) }
+        val localChannel = queryOfflineJob.await()?.also {
             if (pagination.messageFilterDirection == Pagination.LESS_THAN) {
-                updateOldMessagesFromLocalChannel(localChannel)
+                updateOldMessagesFromLocalChannel(it)
             } else {
-                updateLiveDataFromLocalChannel(localChannel)
+                updateLiveDataFromLocalChannel(it)
             }
+            loader.value = false
         }
 
-        // if we are online we we run the actual API call
-        val result = if (queryOnlineJob != null) {
-            val response = queryOnlineJob.await()
-            if (response.isSuccess) {
-                updateLiveDataFromChannel(response.data())
+        val result: Result<Channel> = queryOnlineJob.await().let { onlineResult ->
+            if (onlineResult.isSuccess) {
+                onlineResult.also { updateLiveDataFromChannel(onlineResult.data()) }
             } else {
-                if (response.error().isPermanent()) {
-                    logger.logW("Permanent failure calling channel.watch for channel $cid, with error ${response.error()}")
+                if (onlineResult.error().isPermanent()) {
+                    logger.logW("Permanent failure calling channel.watch for channel $cid, with error ${onlineResult.error()}")
                 } else {
-                    logger.logW("Temporary failure calling channel.watch for channel $cid. Marking the channel as needing recovery. Error was ${response.error()}")
+                    logger.logW("Temporary failure calling channel.watch for channel $cid. Marking the channel as needing recovery. Error was ${onlineResult.error()}")
                     recoveryNeeded = true
                 }
-            }
-            response
-        } else {
-            if (localChannel != null) {
-                Result(localChannel)
-            } else {
-                Result(ChatError("Local channel was not found"))
+                localChannel?.let { Result(it) } ?: onlineResult
```

Comment:
On this branch, `onlineResult` was an error, are we sure we want to return that if we don't have local results either? (Same in the other controller)

Reply:
Yep, We will have a more clear error message on that case
Previously we had:
if online -> onlineResult (success/failure)
else -> if local -> localResult -> else -> Faiulre (Error without any helpful message)